### PR TITLE
Add support for bool data type for condition in where operator

### PR DESCRIPTION
**Pull request by bgawrych**
## Description ##
This change allows to utilize bool data type for condition tensor in oneDNN implementation of where operator

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/operator/nn/dnnl/dnnl_where.cc [hunk 1]
```diff
@@ -34,7 +34,16 @@ namespace op {
 
 // Support for https://oneapi-src.github.io/oneDNN/v2.6/dev_guide_binary.html
 bool SupportDNNLWhere(const std::vector<NDArray>& inputs) {
-  return SupportDNNL<DNNLTypeMode::NoInt32, DNNLTensorsDtypes::Mixed>(inputs);
+  if (inputs[0].dtype() == mshadow::kBool) {
+    // oneDNN natively doesn't support bool data type, however this operator was written
+    // to allow using bool datatype for 'condition' tensor - data will be treated as uint8
+    bool bool_support = SupportDNNLShape<1, 12>(inputs[0].shape());
+    return bool_support &&
+           SupportDNNL<DNNLTypeMode::NoInt32, DNNLTensorsDtypes::AllSame>({inputs[1], inputs[2]});
```
Comment (bartekkuncer) [inline src/operator/nn/dnnl/dnnl_where.cc:42]:
```suggestion
    return SupportDNNLShape<1, 12>(inputs[0].shape()) &&
           SupportDNNL<DNNLTypeMode::NoInt32, DNNLTensorsDtypes::AllSame>({inputs[1], inputs[2]});
```
If you insist on keeping the variable please make it const.

File: src/operator/nn/dnnl/dnnl_where.cc [hunk 2]
```diff
@@ -95,6 +104,25 @@ static mxnet::TShape GetBroadcastableShape(const mxnet::TShape& in_shape,
   return broadcastable_in_shape;
 }
 
+/*!
+ * \brief Create shape vector basing on two input shapes
+ * \param first_shape first input shape
+ * \param second_shape second input shape
+ * \return deducted broadcasted shape basing on first_shape and second_shape
```
Comment (bartekkuncer) [inline src/operator/nn/dnnl/dnnl_where.cc:111]:
```suggestion
 * \return deducted broadcasted shape based on first_shape and second_shape
```

**Review by bartekkuncer — CHANGES_REQUESTED** [review]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [centos-cpu, unix-cpu, unix-gpu]

File: src/operator/nn/dnnl/dnnl_where.cc [hunk 3]
```diff
@@ -34,7 +34,15 @@ namespace op {
 
 // Support for https://oneapi-src.github.io/oneDNN/v2.6/dev_guide_binary.html
 bool SupportDNNLWhere(const std::vector<NDArray>& inputs) {
-  return SupportDNNL<DNNLTypeMode::NoInt32, DNNLTensorsDtypes::Mixed>(inputs);
+  if (inputs[0].dtype() == mshadow::kBool) {
+    // oneDNN natively doesn't support bool data type, however this operator was written
+    // to allow using bool datatype for 'condition' tensor - data will be treated as uint8
```
Comment (bartekkuncer) [inline src/operator/nn/dnnl/dnnl_where.cc:39]: data type or datatype?

**Comment by bgawrych** [comment]
@mxnet-bot run ci [centos-gpu]

**Review by bartekkuncer — APPROVED** [review]

**Pull request merged**